react-accordion/my-app/src/Accordion.tsx

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
+import { Topic } from './App';
+
+type Props = {
+  topics: Topic[];
+};
+
+export function Accordion({ topics }: Props) {
+  const [currentID, setCurrentID] = useState<number>();
+
+  const mappedTopics = topics.map((topic) => (
+    <TopicsCard
+      card={topic}
+      key={topic.id}
+      display={topic.id === currentID}
+      onTitleClick={(target) =>
+        target === currentID ? setCurrentID(undefined) : setCurrentID(topic.id)
+      }
+    />
+  ));
+
+  return <div>{mappedTopics}</div>;
+}
+
+type CardProps = {
+  card: Topic;
+  display?: boolean;
+  onTitleClick: (target: number) => void;
+};
+
+function TopicsCard({ card, display, onTitleClick }: CardProps) {
+  return (
+    <>
+      <h3 onClick={() => onTitleClick(card.id)}>{card.title}</h3>
+      {display ? <p>{card.content}</p> : undefined}
```

Comment:
A slick way to do this is
`{display && <p>{card.content}</p>}`
This works because if `display` is false then short-circuit evaluation replaces the entire expression with `false`, which causes the element to not be displayed. If `display` is true then the expression is replaced with the right-hand side, which is the `p` tag.